test/spoof/spoof.go

```diff
@@ -82,72 +85,75 @@ type SpoofingClient struct {
 	RequestInterval time.Duration
 	RequestTimeout  time.Duration
 
-	endpoint string
-	domain   string
-
 	logf logging.FormatLogger
 }
 
-// New returns a SpoofingClient that rewrites requests if the target domain is not `resolveable`.
+// New returns a SpoofingClient that rewrites requests if the target domain is not `resolvable`.
 // It does this by looking up the ingress at construction time, so reusing a client will not
 // follow the ingress if it moves (or if there are multiple ingresses).
 //
 // If that's a problem, see test/request.go#WaitForEndpointState for oneshot spoofing.
-func New(kubeClientset *kubernetes.Clientset, logf logging.FormatLogger, domain string, resolvable bool, endpointOverride string) (*SpoofingClient, error) {
-	sc := SpoofingClient{
-		Client:          &http.Client{Transport: &ochttp.Transport{Propagation: &b3.HTTPFormat{}}}, // Using ochttp Transport required for zipkin-tracing
-		RequestInterval: requestInterval,
-		RequestTimeout:  RequestTimeout,
-		logf:            logf,
+func New(
+	kubeClientset *kubernetes.Clientset,
+	transport *http.Transport,
+	logf logging.FormatLogger,
+	domain string,
+	resolvable bool,
+	endpointOverride string) (*SpoofingClient, error) {
+	// Spoofing
+	endpoint, err := ResolveEndpoint(kubeClientset, domain, resolvable, endpointOverride)
+	if err != nil {
+		fmt.Errorf("failed get the cluster endpoint: %v", err)
 	}
-
-	var err error
-	if sc.endpoint, err = ResolveEndpoint(kubeClientset, domain, resolvable, endpointOverride); err != nil {
-		return nil, err
+	oldDialContext := transport.DialContext
```

Comment:
Just setup your own. I don't think we can rely that `http.DefaultTransport` is `http.Transport`.

Reply:
What do you mean?
We have to use the DialContext that is provided to us.
We just spoof and delegate to it.

Reply:
What I mean is that you can't rely that this cast will always work. It's an implementation detail of `net/http`. Today it's `http.Transport`, tomorrow it's `http.NewFancyTransport`. Instead just setup up your own dial context (like we do in `networking/transports.go`).

Reply:
Makes sense. Fixed.